shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/controler/DatasetApiController.java

```diff
@@ -398,7 +406,27 @@ else if (dataset instanceof EegDataset) {
 		File zipFile = new File(tmpFile.getAbsolutePath() + ZIP);
 		zipFile.createNewFile();
 
-		zip(tmpFile.getAbsolutePath(), zipFile.getAbsolutePath());
+		Utils.zip(tmpFile.getAbsolutePath(), zipFile.getAbsolutePath());
+		
+		long fileSize = FileUtils.sizeOf(zipFile);
+
+		// Nginx limit is set to 5Gb => split by dataset if file is bigger
+	
+		boolean fileIsTooBig = fileSize > maxFileSize;
+
+		if (fileIsTooBig) {
+			// Publish events to download files on by one
+			
+			for (ShanoirEvent event : events) {
+				// publish all events for every datasets
+				eventService.publishEvent(event);
+			}
+
+			// Here delete zip file only
+			FileUtils.deleteQuietly(zipFile);
+			FileUtils.deleteQuietly(tmpFile);
```

Comment:
Do not delete the file and use it intelligently to be able to re-download the dataset (dataset file by dataset file).
=> Less calculating necessary, and keep it faster.